sklearn/utils/tests/test_validation.py

```diff
@@ -743,7 +743,10 @@ def test_check_array_min_samples_and_features_messages():
         check_array([], ensure_2d=False)
 
     # Invalid edge case when checking the default minimum sample of a scalar
-    msg = r"Singleton array array\(42\) cannot be considered a valid" " collection."
+    msg = (
+        "Expected array-like or collection type, got type `<class 'numpy.ndarray'>` "
```

Comment:
as things stand you can see that the error would be kind of confusing:
- error message: I want an array-like and instead I got a numpy.ndarray. 
- user: but that's what I did I gave you a numpy array and you want an array-like that should be fine no?

Reply:
Oh wow, I totally missed that. :joy: 

What do you think of pulling out the shape like this:

"Expected array-like with shape[0]>0 or or collection type with len(collection)>0, got type `{type(x)}` of shape {x.shape} instead."

Reply:
Calling the wrong type "scalar" would also be possible, but since we don't even tell the user which of the passed in data we mean, I think it's nice to hint to the shapes as well?